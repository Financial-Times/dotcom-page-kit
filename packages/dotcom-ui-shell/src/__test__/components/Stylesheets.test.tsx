import React from 'react'
import renderer from 'react-test-renderer'
import Subject from '../../components/StyleSheets'

describe('dotcom-ui-shell/src/components/StyleSheets', () => {
  it('renders the given stylesheets and critical styles', () => {
    const props = {
      stylesheets: ['path/to/styles.css'],
      criticalStyles: 'html { margin: 0 } body { font-family: "Comic Sans" }'
    }
    const tree = renderer.create(<Subject {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders JavaScript for adding asyncronous stylesheets if appropriate', () => {
    const props = {
      asyncStylesheets: ['path/to/async-styles.css']
    }
    const tree = renderer.create(<Subject {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
