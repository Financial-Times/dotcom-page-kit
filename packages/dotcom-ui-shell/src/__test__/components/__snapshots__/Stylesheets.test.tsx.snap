// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotcom-ui-shell/src/components/StyleSheets renders the given stylesheets and critical styles 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "html { margin: 0 } body { font-family: \\"Comic Sans\\" }",
      }
    }
  />,
  <link
    href="path/to/styles.css"
    rel="stylesheet"
  />,
  <noscript>
    <link
      href="path/to/async.css"
      rel="stylesheet"
    />
  </noscript>,
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "(function loadAsyncStylesheets() {
  var currentScript = document.scripts[document.scripts.length - 1]
  var stylesheets = currentScript.getAttribute('data-stylesheets').split(',')

  for (var i = 0, len = stylesheets.length; i < len; i++) {
    var link = document.createElement('link')
    link.href = stylesheets[i]
    link.key = 'stylesheet-' + stylesheets[i]
    link.rel = 'stylesheet'
    link.media = 'print' // <-- 'print' is intentional; on load, it changes to 'all'.
    link.onload = function (event) {
      event.target.media = 'all'
    }
    currentScript.parentNode.insertBefore(link, currentScript)
  }
})()",
      }
    }
    data-stylesheets="path/to/async.css"
  />,
]
`;
